tests: lib: min_heap: add test coverage for edge cases

kartben opened the pull request:
Add missing tests for edge cases (ex. peek on empty heap, push to full heap, ...) to get to 100% coverage.
cc @sayooj-aerlync

sayoojkkarun reviewed (approved):
Looks good to me. Thanks @kartben

JarmouniA reviewed (dismissed):
May be useful to add test cases for removing & popping from an empty heap as well in another PR.

kartben committed:
tests: lib: min_heap: use more descriptive assert_ macros

make test cases more readable using more direct assert_ macros, for
example zassert_not_null(foo, ...) instead of
zassert_true(foo != NULL, ...).

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
tests: lib: min_heap: add test coverage for edge cases

Add missing tests for edge cases (ex. peek on empty heap, push to full
heap, ...) to get to 100% coverage (lines & branches)

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>
